### PR TITLE
Expand random generator and text tutorial with existing Python sample

### LINKED ISSUE #17312: Sample Python code not included in "Random generator and text with OpenCV" tutorial
The [Random generator and text with OpenCV](https://github.com/opencv/opencv/blob/3.4/doc/tutorials/imgproc/random_generator_and_text/random_generator_and_text.markdown) tutorial only contains C++ code but the program has already been translated to Python and is contained in the OpenCV samples [here](https://github.com/opencv/opencv/blob/3.4/samples/python/drawing.py). (Unfortunately I only noticed this about halfway through creating a translated version of my own!) On a related note, the tutorial also uses in-document code, instead of referring to the actual sample C++ file. If so desired, I would be happy to submit a pull-request for:

- [ ] revising the tutorial to refer to the C++ sample file
- [ ] relocating the Python sample file to the Python tutorial code directory
- [ ] referring to the Python sample file in the tutorial (with toggles for switching between the languages)

### PR DESCRIPTION
**Pull request by dmallia17**
This expands the [Random generator and text with OpenCV](https://docs.opencv.org/3.4/df/d61/tutorial_random_generator_and_text.html) tutorial to include Python coverage, using an existing Python sample file which contains translation of the original C++ source code. 

To accomplish this I followed the steps I identified in the below named issue and also updated the table of contents accordingly. This required adding Doxygen comments in the cpp and py files, and I modified the number of loop iterations in lines(), rectangle() and ellipse() to directly match the C++ source code.  

Closes #17312 
 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: doc/tutorials/imgproc/random_generator_and_text/random_generator_and_text.markdown [hunk 1]
```diff
@@ -125,49 +87,38 @@ Explanation
...
+        -   In the code above we are
+            calling **np.random.randint(a,b)**. This generates a randomly uniformed distribution between
+            the values **a** and **b** (inclusive in **a**, exclusive in **b**).
+        -   From the explanation above, we deduce that the extremes *pt1* and *pt2* will be random
+            values, so the lines positions will be quite impredictable, giving a nice visual effect
+            (check out the Result section below).
+        -   As another observation, we notice that in the @ref cv::line arguments, for the *color*
+            input we generate a random color by using a tuple of three random
+            values generated like so:
+            @snippet samples/python/tutorial_code/imgProc/BasicGeometricDrawing/drawing.py random_color
```
Comment (asmorkalov) [inline doc/tutorials/imgproc/random_generator_and_text/random_generator_and_text.markdown:111]: There are some formatting issue here, lines with different length.

**Comment by dmallia17** [comment]
I just committed again to clean up the lines - I hope I did not make a mistake in consolidating to single lines for those cases!

**Comment by asmorkalov** [comment]
@dmallia17 Please take a look on rendered documentation: https://pullrequest.opencv.org/buildbot/export/pr/17326/docs/df/d61/tutorial_random_generator_and_text.html. There is some formatting issue in explanation section.

**Comment by dmallia17** [comment]
@asmorkalov Apologies but I'm quite confused as to what is happening with this tutorial. Admittedly I am new to Doxygen and I was seeing the same issue (where the toggles and toggled sections don't appear at all) on my end when I built locally, but then again I am also not seeing toggles in the neighboring "Basic Drawing" tutorial when I build locally, so I was hoping it was just  a problem on my side. I can't seem to find what would make the toggles fail to appear and to push so much of the tutorial outside of the regularly formatted section - do you have any intuition as to what might be causing this?

**Comment by asmorkalov** [comment]
It could be some formulae issue or problems with escaping. Let me look on the code in details and make experiments.

**Comment by asmorkalov** [comment]
@alalek Could you look at the PR and generated code? I cannot find issue in the patch and expect that it's some doxygen issue or limitation.

**Comment by alalek** [comment]
I believe `@add_toggle_*` macros can't be used in the lists directly.
Add more empty lines, use `basic_linear_transform.markdown` as a reference.

**Comment by dmallia17** [comment]
Thank you for looking into it! I will try that out very soon; apologies for not doing so sooner but work and some other tasks have taken up my time more recently.

**Comment by asmorkalov** [comment]
@dmallia17 Do you have a chance to finalize the work?

**Comment by dmallia17** [comment]
@asmorkalov Yes, I just spent a good amount of time experimenting with the file and the `basic_linear_transform.markdown`  file and as best I can tell, the bizarre fix is to add in markdown headings, such as the "### Image Processing" heading under the Theory section in the transform file. Only then will the toggles appear, and strangely enough, they appear directly under the heading, not where they are specified in the markdown file.

Finishing the file - various syntax fixes and making sure everything appears as it should - will take some more time (I may be able to finish it this weekend), but at last I'm finally on the right track.

**Comment by dmallia17** [comment]
I have spent more time trying to format the document: while most of the formatting now works with a couple of small fixes, I can't quite get the numbered lists to fully cooperate, possibly because of the toggles interfering. I say this because I was not able to get the toggled material to show correctly _and_ get the numbered list to continue. If there is no objection to removing the numbering, which I don't think hurts readability, I can prepare an update accordingly.

**Comment by alalek** [comment]
https://stackoverflow.com/questions/18088955/markdown-continue-numbered-list

> there is no objection to removing the numbering

No objection.

**Comment by asmorkalov** [comment]
@dmallia17 Does the solution from alalek work for you?

**Comment by dmallia17** [comment]
Unfortunately not - I had found that link and tried that, as well as many other permutations of spaces and newlines, but half the time a number would not show at all, and the other half it would not continue the numbering. I am very sorry but I am going to have to close this pull request, as I am _still_ running into formatting issues (now code snippets are not formatted correctly), and I apologize but I simply do not have the time to continue debugging this. Thank you very much for all of your efforts to help me, but I do not want to continue to leave this pull request open and pending for you, when I am unable to give it my attention.

**Pull request closed without merging**